### PR TITLE
Make env also used while naming provisioner.

### LINKED ISSUE #1168: Unable to change the name of the glusterblock provisioner by setting env PROVISIONER_NAME
It should be possible to start the glusterblock provisioner with the name of user's choice by setting only the env PROVISIONER_NAME. 

Currently, it is does not work.

### PR DESCRIPTION
**Pull request by humblec**
Fixes #1168

Signed-off-by: hchiramm <hchiramm@redhat.com>

### CODE REVIEW

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go [hunk 1]
```diff
@@ -905,12 +905,10 @@ func main() {
 	provName := provisionerName
 	provEnvName := os.Getenv(provisionerNameKey)
 
-	// Precedence is given for ProvisionerNameKey
-	if provEnvName != "" && *id != "" {
+	// Precedence is given for ProvisionerNameKey if both are set
+	if provEnvName != "" && *id != "" || provEnvName != "" {
```
Comment (raghavendra-talur) [inline gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: The logic currently reads like this:

```
if (a && b) || a {
  use ENV
} else if b {
  use cli
}
```

If we want to keep logic as in the table 

| ENV set | CLI set | Name of the provisioner|
| ------------- | ------------- | ------------- |
| No | No | Default |
| Yes | No | ENV |
| No | Yes | CLI |
| Yes | Yes | ENV |

Then I suggest we simplify the code as
```
if a {
  use ENV
} else if b {
  use cli
}
```

**Comment by humblec** [comment]
>Then I suggest we simplify the code as
>
@raghavendra-talur  Thats for sure, I deliberately kept that part `( a && b)` for better code reading, not really for logic. but, I can remove it and ONLY keep source code comment if you prefer.  Please let me know your choice.

@phlogistonjohn  @raghavendra-talur  Coming back to the priority, I am still not sure or cant `100%` confirm what should get priority between `cli` and `env` , may be there are different opinions and dont have any writen rule. However I am really worried  about backward compatibility which is a must thing assured in kubernetes. So, lets give preference to `ENV` to make sure we are not breaking.

**Comment by humblec** [comment]
@raghavendra-talur @phlogistonjohn any comments or thoughts ?

**Comment by humblec** [comment]
@raghavendra-talur if we are agreed on preference part, we can change https://github.com/kubernetes-incubator/external-storage/pull/1169 or we can take this PR in. I am fine with any.

**Comment by humblec** [comment]
@wongma7 PTAL.

**Comment by humblec** [comment]
@wongma7 can you please merge this PR ?

**Pull request merged**